plugin/sqlsh/src/main/java/com/twosigma/beaker/sqlsh/utils/ConnectionStringHolder.java

```diff
@@ -0,0 +1,120 @@
+package com.twosigma.beaker.sqlsh.utils;
+
+import java.util.StringTokenizer;
+
+public class ConnectionStringHolder {
+
+  public static final String USER_CONNECTION_KEY = "user";
+  public static final String PASSWORD_CONNECTION_KEY = "password";
+  public static final String KEY_FOR_DISPLAY_INPUT_DIALOG = "<prompt>";
+  public static final String EQUAL_SIGN = "=";
+  public static final String QUESTION_SIGN = "?";
+  public static final String AND_SIGN = "&";
+	
+  private String connectionString;
+  private String user;
+  private String password;
+  private boolean showDialog;
+  
+  public ConnectionStringHolder(String connectionString){
+  	setConnectionString(connectionString);
+  }
+	
+	public String getUser() {
+		return user;
+	}
+	public void setUser(String user) {
+		this.user = user;
+	}
+	public String getPassword() {
+		return password;
+	}
+	public void setPassword(String password) {
+		this.password = password;
+	}
+	public boolean isShowDialog() {
+		return showDialog;
+	}
+	public void setShowDialog(boolean showDialog) {
+		this.showDialog = showDialog;
+	}
+	public String getConnectionString() {
+		return connectionString;
+	}
+	
+	public void setConnectionString(String connectionString) {
+		int beginIndex = connectionString.indexOf(QUESTION_SIGN);
```

Comment:
Hard to use standart library because (Tried us URL and URI):
1) "<prompt>" --- gives  "Exception in thread "main" java.net.URISyntaxException: Illegal character in opaque part at index 65:"
2) "jdbc:postgresql:" gives : Exception in thread "main" java.net.MalformedURLException: unknown protocol: jdbc